middleware/afc_status.py

```
@@ -67,13 +107,10 @@ def _publish_lane_actions(lane_name: str, action: str | None, pending: dict | No
 
     if newly_loaded and pending:
-        logger.info(f"AFC {source}: {lane_name} just loaded — sending cached tag data")
-        _send_afc_lane_data(
-            lane_name,
-            pending.get("color_hex", ""),
-            pending.get("material", ""),
-            pending.get("remaining_g"),
-        )
+        # Delayed 2s to let AFC's own load sequence finish first — if sent too early,
+        # AFC overwrites material/weight during its load process.
+        logger.info(f"AFC {source}: {lane_name} just loaded — scheduling lane data (2s delay)")
```

Ruff (0.15.9)

[warning] Logging statement uses f-string

(G004)